src/API/Site/Controllers/MerchantCenter/AttributeMappingRulesController.php

```diff
@@ -0,0 +1,255 @@
+	 *
+	 * @return callable
+	 */
+	protected function delete_mapping_rule_callback(): callable {
+		return function( Request $request ) {
+			try {
+				$rule_id = absint( $request->get_param( 'id' ) );
+
+				if ( ! $this->attribute_mapping_rules_query->delete( 'id', $rule_id ) ) {
+					return $this->response_from_exception( new Exception( 'Unable to delete the rule' ) );
+				}
+
+				return [
+					'id' => $rule_id,
+				];
+			} catch ( Exception $e ) {
+				return $this->response_from_exception( $e );
+			}
+		};
+	}
+
+
+	/**
+	 * Get the item schema properties for the controller.
+	 *
+	 * @return array The Schema properties
+	 */
+	protected function get_schema_properties(): array {
+		return [
+			'id'                      => [
+				'description'       => __( 'The Id for the rule.', 'google-listings-and-ads' ),
+				'type'              => 'integer',
+				'validate_callback' => 'rest_validate_request_arg',
+				'readonly'          => true,
+			],
+			'attribute'               => [
+				'description'       => __( 'The attribute value for the rule.', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'validate_callback' => 'rest_validate_request_arg',
+				'required'          => true,
+				'enum'              => array_column( $this->attribute_mapping_helper->get_attributes(), 'id' ),
+			],
+			'source'                  => [
+				'description'       => __( 'The source value for the rule.', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'validate_callback' => 'rest_validate_request_arg',
+				'required'          => true,
+			],
+			'category_condition_type' => [
+				'description'       => __( 'The category condition type to apply for this rule.', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'validate_callback' => 'rest_validate_request_arg',
+				'required'          => true,
+				'enum'              => $this->attribute_mapping_helper->get_category_condition_types(),
+			],
+			'categories'              => [
+				'description'       => __( 'Comma separated categories for this rule.', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'required'          => false,
+				'validate_callback' => function( $param ) {
+					return $this->validate_categories_param( $param );
+				},
+			],
+		];
+	}
+
+	/**
+	 * Get the item schema name for the controller.
+	 *
+	 * Used for building the API response schema.
+	 *
+	 * @return string
+	 */
+	protected function get_schema_title(): string {
+		return 'attribute_mapping_rules';
+	}
+
+	/**
+	 * @param string $categories  Categories to validate
+	 * @return bool|WP_Error  True if it's validated
+	 *
+	 * @throw Exception when invalid categories are provided
+	 */
+	public function validate_categories_param( string $categories ) {
+			$categories_array = explode( ',', $categories );
+
+		foreach ( $categories_array as $category ) {
+			if ( ! is_numeric( $category ) ) {
```

Comment:
In theory the categories tree this will come from another API call rendering a TreeSelectControl with the category IDs saved as an array. Then I will use `join(,)` function (the JS implode equivalent). So I guess it's safe.